src/server/worker.cc

```diff
@@ -76,17 +78,29 @@ Worker::Worker(Server *srv, Config *config) : srv(srv), base_(event_base_new())
 }
 
 Worker::~Worker() {
-  std::vector<redis::Connection *> conns;
-  conns.reserve(conns_.size() + monitor_conns_.size());
-
-  for (const auto &iter : conns_) {
-    conns.emplace_back(iter.second);
-  }
-  for (const auto &iter : monitor_conns_) {
-    conns.emplace_back(iter.second);
+  auto collect_conns_fn = [](ConnMap &conns) {
+    return parallel_reduce(
```

Comment:
Is `parallel_reduce` necessary here?

Reply:
Hi @PragmaTwice ,

I use a reduce action before iteration because the `parallel_for` will acquire a range rather than one item.
So, I want to move the time costly action `Close()` outside of the `parallel_for`.

But since this function is inside the dtor, if the ownership is managed correctly, it should be the only function uses internal state of the `Worker` object.
I think the `parallel_reduce` can be replaced by `parallel_for`.

I will refactor it and run tests before pushing to current PR.

Best Regards,
Edward

Reply:
I mean maybe a single-thread for is enough here? didn't have a deep look yet cc @mapleFU

Reply:
Hi @PragmaTwice ,

I have removed the `parallel_reduce` from dtor and all tests are passed in my repo's GitHub Actions.
I think in dtor the concurrency protection of internal state may be redundant since all references should be released except the dtor itself.

Best Regards,
Edward

Reply:
> I mean maybe a single-thread for is enough here? didn't have a deep look yet

Yeah parallel task also has a dop (degree of parallel), and I guess default of `parallel_for` would be high and be good at writing cpu-bound tasks like OpenMP

I think maybe lock is prefered here